pdp/handlers.go

```diff
@@ -883,6 +883,133 @@ func (p *PDPService) handleAddRootToProofSet(w http.ResponseWriter, r *http.Requ
+		http.Error(w, "Missing root ID in URL", http.StatusBadRequest)
+		return
+	}
+
+	// Convert proofSetId to uint64
+	proofSetID, err := strconv.ParseUint(proofSetIdStr, 10, 64)
+	if err != nil {
+		http.Error(w, "Invalid proof set ID format", http.StatusBadRequest)
+		return
+	}
+	rootID, err := strconv.ParseUint(rootIdStr, 10, 64)
+	if err != nil {
+		http.Error(w, "Invalid root ID format", http.StatusBadRequest)
+		return
+	}
+
+	// check if the proofset belongs to the service in pdp_proof_sets
+	var proofSetService string
+	err = p.db.QueryRow(ctx, `
+			SELECT service 
+			FROM pdp_proof_sets
+			WHERE id = $1
+		`, proofSetID).Scan(&proofSetService)
+	if err != nil {
+		if err == pgx.ErrNoRows {
+			http.Error(w, "Proof set not found", http.StatusNotFound)
+			return
+		}
+		http.Error(w, "Failed to retrieve proof set: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if proofSetService != serviceLabel {
+		// same as when actually not found to avoid leaking information in obvious ways
+		http.Error(w, "Proof set not found", http.StatusNotFound)
+		return
+	}
+
+	// Get the ABI and pack the transaction data
+	abiData, err := contract.PDPVerifierMetaData.GetAbi()
+	if err != nil {
+		http.Error(w, "Failed to get contract ABI: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Pack the method call data
+	data, err := abiData.Pack("scheduleRemovals",
+		big.NewInt(int64(proofSetID)),
+		[]*big.Int{big.NewInt(int64(rootID))},
+		[]byte{},
+	)
+	if err != nil {
+		http.Error(w, "Failed to pack method call: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Get the sender address
+	fromAddress, err := p.getSenderAddress(ctx)
+	if err != nil {
+		http.Error(w, "Failed to get sender address: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Prepare the transaction
+	ethTx := types.NewTransaction(
+		0, // nonce will be set by SenderETH
+		contract.ContractAddresses().PDPVerifier,
+		big.NewInt(0), // value
+		0,             // gas limit (will be estimated)
+		nil,           // gas price (will be set by SenderETH)
+		data,
+	)
+
+	// Send the transaction
+	reason := "pdp-delete-root"
+	txHash, err := p.sender.Send(ctx, fromAddress, ethTx, reason)
+	if err != nil {
+		http.Error(w, "Failed to send transaction: "+err.Error(), http.StatusInternalServerError)
+		log.Errorf("Failed to send transaction: %+v", err)
+		return
+	}
+
+	// Schedule deletion of the root from the proof set using a transaction
+	_, err = p.db.BeginTransaction(ctx, func(tx *harmonydb.Tx) (bool, error) {
+		// Insert into message_waits_eth
+		_, err := tx.Exec(`
+			INSERT INTO message_waits_eth (signed_tx_hash, tx_status)
+			VALUES ($1, $2)
+		`, txHash.Hex(), "pending")
+		if err != nil {
+			return false, err
+		}
+
+		return true, nil
+	}, harmonydb.OptionRetry())
+
+	if err != nil {
+		if err.Error() == "root not found" {
+			http.Error(w, "Root not found", http.StatusNotFound)
+			return
+		}
```

Comment:
The insert is non-returning, this check doesn't really make sense to me

Reply:
My bad this was a relic of the design where I tracked the pending deletes in a new table